ecommerce/serializers.py

```diff
@@ -227,6 +232,36 @@ class Meta:
         model = models.Coupon
 
 
+def _serialize_item(*, item, basket, context):
+    """
+    Serialize a BasketItem
+
+    Args:
+        item (BasketItem): A basket item
+        basket (Basket): A basket
+        context (dict): Context from the BasketSerializer
+
+    Returns:
+        dict:
+            A serialized representation
+    """
+    serialized_product_version = ProductVersionSerializer(
+        instance=latest_product_version(item.product), context=context
+    ).data
+    valid_run_ids = set()
+    for course in serialized_product_version["courses"]:
+        for run in course["courseruns"]:
+            valid_run_ids.add(run["id"])
+    run_ids = [
+        run_id
+        for run_id in models.CourseRunSelection.objects.filter(
+            basket=basket
+        ).values_list("run", flat=True)
+        if run_id in valid_run_ids
+    ]
```

Comment:
I think this can be rewritten as:

```python
    run_ids = models.CourseRunSelection.objects.filter(
        basket=basket, run_id__in=valid_run_ids
    ).values_list("run", flat=True)
```